ocaml/xapi/xapi_vbd.ml

```diff
@@ -77,9 +77,14 @@ let plug ~__context ~self =
   let domid = Int64.to_int (Db.VM.get_domid ~__context ~self:vm) in
   let force_loopback_vbd = Helpers.force_loopback_vbd ~__context in
   let hvm = Helpers.has_qemu_currently ~__context ~self:vm in
-  if
-    System_domains.storage_driver_domain_of_vbd ~__context ~vbd:self = vm
-    && not force_loopback_vbd
+  if domid = 0
+  then (
+    (* SDS vm support *)
+    Xapi_xenops.vbd_plug ~__context ~self
```

Comment:
Can we merge this with the `else` branch below, which also does `Xapi_xenops.vbd_plug` plus some other checks that may be relevant here as well?